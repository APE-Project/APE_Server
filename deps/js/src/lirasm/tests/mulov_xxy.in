; 1073741824 * 2 >= 2^31, and will overflow.
big = immi 1073741824
two = immi 2

; Because 'big' isn't used after mul, it _may_ get allocated to the same
; register as 'res'. This is the case with the ARM back-end, and that is where
; this test is important as rX=rX*rY isn't possible on ARMv5 without some
; trickery.
res = mulxovi big two    ; overflow, so we exit here

; Store 'res' so it isn't dead.
m = allocp 4
sti res m 0
x                       ; we don't exit here
