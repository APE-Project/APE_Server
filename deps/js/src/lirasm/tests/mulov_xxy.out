Exited block on line: 9
